source/client/factories_impl.cc

```diff
@@ -126,5 +129,32 @@ HeaderSourcePtr HeaderSourceFactoryImpl::create() const {
   return std::make_unique<StaticHeaderSourceImpl>(std::move(header));
 }
 
+TerminationPredicateFactoryImpl::TerminationPredicateFactoryImpl(const Options& options)
+    : OptionBasedFactoryImpl(options) {}
+
+TerminationPredicatePtr
+TerminationPredicateFactoryImpl::create(Envoy::TimeSource& time_source, Envoy::Stats::Scope& scope,
+                                        const Envoy::MonotonicTime start) const {
+  // By default, we don't tolerate status code errors & connection-level errors.
```

Comment:
Is this new behavior? There already are a bunch of folks who are getting used to Nighthawk. If we are going to start changing things like this, we probably need a version history file that tracks this kind of breaking change.

Reply:
Yes this is new behavior. We take responsibility for determining success/failure based on the termination predicates.
I think these defaults represent an improvement, but backing it out is easy if it causes more trouble then it is worth.

Reply:
I think it's fine to do this, but can you add a file similar to https://github.com/envoyproxy/envoy/blob/master/docs/root/intro/version_history.rst (it can just be a simple Markdown for now) that tracks changes like this? Ideally we update it as we go, and then we have version release notes.

Reply:
Sure, updated that file for things related to this PR, I'll update it some more in a follow up to include earlier work.